lib/nerve/rate_limiter.rb

```diff
@@ -0,0 +1,45 @@
+module Nerve
+  # RateLimiter implements a TokenBucket-based rate-limiting strategy.
+  # It allows an average of `average_rate` tokens per `period`, with a
+  # maximum burst of `max_burst` when the bucket is full.
+  # `period` is the time period for `average_rate`, in seconds.
+  # Note: a single instance of RateLimiter is *not* thread-safe.
+  # See: https://en.wikipedia.org/wiki/Token_bucket
+  class RateLimiter
+    def initialize(average_rate: Float::INFINITY, max_burst: Float::INFINITY)
+      raise ArgumentError, "average_rate should be numeric" unless average_rate.is_a? Numeric
```

Comment:
`average_rate` should be > 0.0

Reply:
I'll put in a check for `>= 0.0`; it's possible that in some case you want to entirely prevent any requests from going through, in which case a rate of 0 would be useful